pkg/models/load_balancer.go

```diff
@@ -138,22 +145,92 @@ type CreateLBMonitor struct {
 }
 
 type CreateLBMonitorReq struct {
-	ID                 int    `json:"-" tf:"id,computed"`
-	Name               string `json:"name"`
-	Description        string `json:"description"`
-	MonitorType        string `json:"monitorType" tf:"monitor_type"`
-	MonitorTimeout     int    `json:"monitorTimeout" tf:"monitor_timeout"`
-	MonitorInterval    int    `json:"monitorInterval" tf:"monitor_interval"`
-	SendVersion        string `json:"sendVersion" tf:"send_version"`
-	SendType           string `json:"sendType" tf:"send_type"`
-	ReceiveCode        string `json:"receiveCode" tf:"receive_code"`
-	MonitorDestination string `json:"monitorDestination" tf:"monitor_destination"`
-	MonitorReverse     bool   `json:"monitorReverse" tf:"monitor_reverse"`
-	MonitorTransparent bool   `json:"monitorTransparent" tf:"monitor_transparent"`
-	MonitorAdaptive    bool   `json:"monitorAdaptive" tf:"monitor_adaptive"`
-	FallCount          int    `json:"fallCount" tf:"fall_count"`
-	RiseCount          int    `json:"riseCount" tf:"rise_count"`
-	AliasPort          int    `json:"aliasPort" tf:"alias_port"`
+	ID                  int                         `json:"-" tf:"id,computed"`
+	LbID                int                         `json:"-" tf:"lb_id"`
+	Name                string                      `json:"name" tf:"name"`
+	Description         string                      `json:"description" tf:"description"`
+	Type                string                      `json:"monitorType" tf:"type"`
+	Timeout             int                         `json:"monitorTimeout"`
+	Interval            int                         `json:"monitorInterval"`
+	RequestVersion      string                      `json:"sendVersion"`
+	RequestMethod       string                      `json:"sendType"`
+	ResponseStatusCodes string                      `json:"receiveCode"`
+	ResponseData        string                      `json:"receiveData"`
+	RequestURL          string                      `json:"monitorDestination"`
+	RequestBody         string                      `json:"sendData"`
+	AliasPort           int                         `json:"aliasPort"`
+	RiseCount           int                         `json:"riseCount"`
+	FallCount           int                         `json:"fallCount"`
+	DataLength          int                         `json:"dataLength"`
+	MaxFail             int                         `json:"maxRetry"`
+	TfHttpConfig        *CreateHttpMonitorConfig    `json:"-" tf:"http_monitor,sub"`
+	TfHttpsConfig       *CreateHttpsMonitorConfig   `json:"-" tf:"https_monitor,sub"`
+	TfIcmpConfig        *CreateIcmpMonitorConfig    `json:"-" tf:"icmp_monitor,sub"`
+	TfPassiveConfig     *CreatePassiveMonitorConfig `json:"-" tf:"passive_monitor,sub"`
+	TfTcpConfig         *CreateTcpMonitorConfig     `json:"-" tf:"tcp_monitor,sub"`
+	TfUdpConfig         *CreateUdpMonitorConfig     `json:"-" tf:"udp_monitor,sub"`
+}
+
+type CreateHttpMonitorConfig struct {
```

Comment:
JSON is not needed here ?

Reply:
no need of json because these models CreateHttpMonitorConfig only to achieve the monitor type payloads from terraform later to send to cmp we are using same flat models.

Reply:
where are we having keys for json then ?